### PR TITLE
add permissions to the graph drives

### PR DESCRIPTION
**Pull request by C0rby**
## Description

Added permissions to graph drives when listing drives.
A drive looks like this after this change:
```json
{
    "driveType": "personal",
    "id": "058bff95-6708-4fe5-91e4-9ea3d377588b",
    "lastModifiedDateTime": "2022-02-03T11:15:11.464382312+01:00",
    "name": "Maurice Moss",
    "owner": {
        "user": {
            "id": "058bff95-6708-4fe5-91e4-9ea3d377588b"
        }
    },
    "root": {
        "id": "058bff95-6708-4fe5-91e4-9ea3d377588b",
        "permissions": [
            {
                "grantedTo": [
                    {
                        "user": {
                            "id": "058bff95-6708-4fe5-91e4-9ea3d377588b"
                        }
                    }
                ],
                "roles": [
                    "manager"
                ]
            }
        ],
        "webDavUrl": "https://localhost:9200/dav/spaces/058bff95-6708-4fe5-91e4-9ea3d377588b"
    }
}
```

## Motivation and Context

We want to see who has access to a space.

## Types of changes

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] Code changes

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>localApiTests-apiSpaces-ocis</strong>](https://drone.owncloud.com/owncloud/ocis/9176/20/1) failed. Further test are cancelled...

File: graph/pkg/service/v0/drives.go [hunk 1]
```diff
@@ -406,21 +406,78 @@ func (g Graph) ListStorageSpacesWithFilters(ctx context.Context, filters []*stor
...
+		}
+		if len(m) != 0 {
+			managerIdentities := []libregraph.IdentitySet{}
+			editorIdentities := []libregraph.IdentitySet{}
+			viewerIdentities := []libregraph.IdentitySet{}
+
+			for id, perm := range m {
+				// This temporary variable is necessary since we need to pass a pointer to the
+				// libregraph.Identity and if we pass the pointer from the loop every identity
+				// will have the same id.
```
Comment (micbar) [inline graph/pkg/service/v0/drives.go:437]: nice workaround

File: graph/pkg/service/v0/drives.go [hunk 2]
```diff
@@ -406,21 +406,78 @@ func (g Graph) ListStorageSpacesWithFilters(ctx context.Context, filters []*stor
...
+				tmp := id
+				identity := libregraph.IdentitySet{User: &libregraph.Identity{Id: &tmp}}
+				switch {
+				case perm.AddGrant:
```
Comment (micbar) [inline graph/pkg/service/v0/drives.go:441]: so `AddGrant` is our reverse engineering criterium for the manager role.

**Review by micbar — APPROVED** [review]
LGTM, let us add an odata query later

**Review by kobergj — APPROVED** [review]
Looks good! Just wondering why the changes to `expected-failures` and `go.mod` still appear. They are already in master :smile:

File: graph/pkg/service/v0/drives.go (hunk 2 above)
Comment (C0rby) [reply graph/pkg/service/v0/drives.go:441]: Yes, exactly. That's currently the only way.

**Pull request merged**